benchmark/testsystems/bookkeepers.py

```diff
@@ -45,138 +45,114 @@ def accumulate_shadow_work(self, x_0, v_0, n_steps):
 
         # Collect equilibrium samples
         print("Collecting equilibrium samples...")
         equilibrium_samples = []
+        box_vectors = []
         for _ in tqdm(range(self.n_samples)):
             self.unbiased_simulation.step(self.thinning_interval)
-            x = self.unbiased_simulation.context.getState(getPositions=True).getPositions(asNumpy=True)
+            state = self.unbiased_simulation.context.getState(getPositions=True)
+            x = state.getPositions(asNumpy=True)
             equilibrium_samples.append(strip_unit(x))
-        print("Equilibrated GHMC acceptance rate: {:.3f}%".format(100 * self.get_ghmc_acceptance_rate()))
+            box_vectors.append(state.getPeriodicBoxVectors())
+        print("Equilibrated XC-GHMC acceptance rate: {:.3f}%".format(100 * self.get_acceptance_rate()))
 
-        return np.array(equilibrium_samples)
+        return np.array(equilibrium_samples), np.array(box_vectors)
 
-    def get_path_to_samples(self):
+    def get_path_to_samples_and_box_vectors(self):
         """Samples are {name}_samples.npy in DATA_PATH"""
-        return os.path.join(DATA_PATH, '{}_samples.npy'.format(self.name))
+        samples = os.path.join(DATA_PATH, '{}_samples.npy'.format(self.name))
+        box_vectors = os.path.join(DATA_PATH, '{}_box_vectors.npy'.format(self.name))
+        return samples, box_vectors
 
     def check_for_cached_samples(self):
         """Check if there's a file where we expect to find cached
         equilibrium samples.
         """
         # TODO : Need to check if any of the simulation parameters have changed.
-        return os.path.exists(self._path_to_samples)
+        return os.path.exists(self._path_to_samples) and os.path.exists(self._path_to_box_vectors)
 
-    def save_equilibrium_samples(self, x_samples):
+    def save_equilibrium_samples(self, x_samples, box_vectors):
         """Save numpy archive of equilibrium samples to disk."""
         np.save(self._path_to_samples, x_samples)
+        np.save(self._path_to_box_vectors, box_vectors)
 
     def load_equilibrium_samples(self):
         """Load numpy archive of equilibrium samples"""
         print("Loading equilibrium samples from {}...".format(self._path_to_samples))
         x_samples = np.load(self._path_to_samples)
-        return x_samples
+        print("Loading box vectors from {}...".format(self._path_to_box_vectors))
+        x_box_vectors = np.load(self._path_to_box_vectors)
+        return x_samples, x_box_vectors
 
     def sample_x_from_equilibrium(self):
         """Draw sample (uniformly, with replacement) from cache of configuration samples"""
         if self.cached == False:
             self.load_or_simulate_x_samples()
 
-        return self.x_samples[np.random.randint(len(self.x_samples))]
+        i = np.random.randint(len(self.x_samples))
+        return self.x_samples[0][i], self.x_samples[1][i]
```

Comment:
I'm not sure I understand what `x_samples[0][i]` and `x_samples[1][i]` are. Can you document your return values in the docstring headers?